appcues/src/main/java/com/appcues/statemachine/StateMachine.kt

```diff
@@ -77,8 +78,11 @@ internal class StateMachine(
         lifecycleTracker.start(this, { onEndedExperience?.invoke(it) })
     }
 
+    @OptIn(ExperimentalCoroutinesApi::class)
     private fun stopLifecycleTracking() {
         lifecycleTracker.stop()
+        _stateFlow.resetReplayCache()
+        _errorFlow.resetReplayCache()
```

Comment:
this is the semi-related fix, so that when multiple experiences are attempted due to failure/fallback - the subsequent attempts dont re-emit the errors from the previous try, due to the SharedFlow having a retry: 1 (avoiding changing that for now to hopefully not introduce any unexpected regressions elsewhere)